atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/DefaultSweepStateCoordinator.java

```diff
@@ -134,34 +135,34 @@ private Optional<LockedItem<SweepableBucket>> getFirstUnlockedBucket(Stream<Lock
     }
 
     private void updateBuckets(Set<SweepableBucket> newBuckets) {
-        Map<Integer, List<SweepableBucket>> partition = newBuckets.stream()
+        Map<ShardAndStrategy, List<SweepableBucket>> partition = newBuckets.stream()
                 .sorted(SweepableBucketComparator.INSTANCE)
-                .collect(Collectors.groupingBy(
-                        bucket -> bucket.bucket().shardAndStrategy().shard()));
+                .collect(Collectors.groupingBy(bucket -> bucket.bucket().shardAndStrategy()));
 
-        List<Lockable<SweepableBucket>> firstBucketsOfEachShard = partition.values().stream()
+        List<Lockable<SweepableBucket>> firstBucketsOfEachShardAndStrategy = partition.values().stream()
                 .filter(list -> !list.isEmpty())
                 .map(list -> list.get(0))
                 .map(lockableFactory::createLockable)
-                .collect(Collectors.toUnmodifiableList());
+                .toList();
 
         List<Lockable<SweepableBucket>> remainingBuckets = partition.values().stream()
                 .flatMap(list -> list.stream().skip(1))
                 .map(lockableFactory::createLockable)
-                .collect(Collectors.toUnmodifiableList());
+                .toList();
 
         // There's a delay between setting each variable, but we do not require (for correctness) that these two
         // variables are updated atomically.
 
         bucketsLists = ImmutableBucketsLists.builder()
-                .firstBucketsOfEachShard(firstBucketsOfEachShard)
+                .firstBucketsOfEachShardAndStrategy(firstBucketsOfEachShardAndStrategy)
                 .remainingBuckets(remainingBuckets)
                 .build();
         seenBuckets = ConcurrentHashMap.newKeySet();
         log.info(
-                "Updated sweepable buckets (first buckets per shard: {}, remaining buckets: {}).",
-                SafeArg.of("firstBucketsOfEachShard", firstBucketsOfEachShard),
-                SafeArg.of("remainingBuckets", remainingBuckets));
+                "Updated sweepable buckets (number of first buckets per shard and strategy: {},"
+                        + " number of remaining buckets: {}).",
+                SafeArg.of("firstBucketsOfEachShardAndStrategySize", firstBucketsOfEachShardAndStrategy.size()),
```

Comment:
Drive by to reduce log size - see the logging section in the questions above.

I'm open to logging the buckets themselves if debug logs are enabled. That will be a different PR though.